contracts/HybridCustody.cdc

```diff
@@ -402,6 +422,20 @@ pub contract HybridCustody {
             return self.childCap.borrow()!.borrowAccount().borrow<&CapabilityProxy.Proxy>(from: StoragePath(identifier: path)!)
         }
 
+        pub fun getViews(): [Type] {
```

Comment:
I agree, we'll want to think about how to support various types of views that may emerge in the future. A couple ways I can think of doing this so long as we're allowing for user-defined metadata:

1. Define a minimum set of a metadata values and define them in an interface. We then set the type of the metadata variable (in this case `data`) to that standardized interface - see [this implementation](https://github.com/onflow/linked-accounts/blob/main/contracts/LinkedAccounts.cdc#L81) for reference. This approach gives us flexibility while also guaranteeing metadata will at minimum include certain fields.
2. Maintain a mapping of `data: {Type: AnyStruct}` where views are indexed on their Type. While more storage intensive, we get the flexibility of adding new view types and `getViews()` and `resolveView()` are easy enough to implement by returning keys and values of the mapping respectively.
3. Define a resolver struct interface and add it as a field on `ProxyAccount`. This would enable arbitrary view definitions within the implementing struct, but we lose guarantees that we will be able to amend those views. An example of that [here](https://github.com/onflow/linked-accounts/blob/main/contracts/LinkedAccounts.cdc#L83)

Reply:
If it helps, I'll share the implementation our team has taken for NFTs. [MetadataViewsManager](https://github.com/Niftory/niftory-flow/blob/main/cadence/contracts/MetadataViewsManager.cdc) allows you to create a resource that lets you add and remove views. You can define the actual view implementations separately like [this](https://github.com/Niftory/niftory-flow/blob/main/cadence/contracts/NiftoryMetadataViewsResolvers.cdc). Then `getViews` and `resolveView` can look something like [this](https://github.com/Niftory/niftory-flow/blob/main/cadence/contracts/NiftoryTemplate.cdc#L161). The manager can also be locked and allows for any arbitrary data from the contract calling `resolveView` (a public reference to the NFT in our case).

Reply:
Thanks @siddthesquid! I'll take a look at these today, I remember @an1va mentioning them to me some time back

Reply:
This system makes sense to me, I'll see what I can do to add something like it. 

My initial thinking would be to register a type -> `{Resolver}` mapping which the the owner of a given account can add to. The question, really, is what a parent can change, and what the child account can change.

That is why display is its own field on the ProxyAccount resource. If we want to consolidate, we'd probably want to distinguish between a parent-managed view, and an owner-managed view which can get quite tricky when you think about how to adhere to the MetadataViews Standard